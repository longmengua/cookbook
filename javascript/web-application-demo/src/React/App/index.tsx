import React from "react";
import "./index";
import "./index_c";

function App() {
	return (
		<div className="App">
			Index start
		</div>
	);
}

export default App;
