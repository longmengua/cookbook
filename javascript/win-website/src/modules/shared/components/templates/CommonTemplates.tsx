import Header from "../atomic/Header";
import Footer from "../atomic/Footer";
import React from "react";

class CommonTemplates extends React.Component {
    public render(): JSX.Element {
        return (
            <>
                <Header />
                {this.props.children}
                <Footer />
            </>
        );
    }
}
export default CommonTemplates;
