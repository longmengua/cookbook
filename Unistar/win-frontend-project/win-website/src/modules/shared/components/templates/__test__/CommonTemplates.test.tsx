import "jest-enzyme";
import CommonTemplates from "../CommonTemplates";
import {shallow} from "enzyme";
import React from "react";

describe("<CommonTemplates />", (): void => {
    it("take a snapshot", (): void => {
        const wrapper = shallow(<CommonTemplates />);
        expect(wrapper).toMatchSnapshot();
    });
});
