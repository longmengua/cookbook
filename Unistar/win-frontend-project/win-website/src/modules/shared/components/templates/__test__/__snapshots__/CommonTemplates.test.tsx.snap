// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommonTemplates /> take a snapshot 1`] = `
<Fragment>
  <withRouter(Header) />
  <Footer />
</Fragment>
`;
